### PR TITLE
Code Coverage support

### LINKED ISSUE #8: How to get code coverage (coveralls?) going?
Hey there, I'm not sure if here's the right place to raise this as an issue, but I'm wondering how to get code coverage reporting going with huginn_agent

Maybe some background, I'm not a ruby coder, yet love Huginn and recently wrote a [Todoist Agent](https://github.com/stesie/huginn_todoist_agent) for it. Yet simple, it just allows creating items there (querying still left on todo)

And as Huginn has this hook to report to Coveralls and the agent's specs are run over Huginn it immediately wanted to report to there (but failed with "Couldn't find a repository matching this job." as I didn't register the repo there). So I registered to Coveralls and tried again, ... and reporting is green now.

... yet coverage is wrong, it reports 100% (yay!) but zero files covered (meh). 

This is also reproducible locally:

```console
[stesie@peer 130 ~/Projekte/huginn_todoist_agent master|✔ ]$ export COVERAGE=1
[stesie@peer 0 ~/Projekte/huginn_todoist_agent master|✔ ]$ bundler exec rake spec
Running specs ...
Run options: include {:focus=>true}

All examples were filtered out; ignoring {:focus=>true}

Randomized with seed 5083
..................

Finished in 2.15 seconds (files took 4.65 seconds to load)
18 examples, 0 failures

Randomized with seed 5083

Coverage report generated for RSpec to /home/stesie/Projekte/huginn_todoist_agent/spec/huginn/coverage.
0.0 / 0.0 LOC (100.0%) covered.
[OK]
```

I suppose this is due to specs + subject under test are outside of the actual source tree (i.e. `../../`), but don't know what (best) to do about that

### PR DESCRIPTION
**Pull request by stesie**
Here's the pull request as discussed yesterday.

Both local code coverage (with simplecov enabled via COVERAGE=1) as well as coveralls reporting on travis are working fine.

Fixes #8

### CODE REVIEW

**Comment by dsander** [comment]
Nice! Thanks you for testing it on Travis @stesie 

@cantino Are you okay with with monkey patch?

File: lib/huginn_agent/patches/coverage.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'simplecov'
+
+if !ENV['COVERAGE']
+  require 'coveralls'
+  module Coveralls
+    module Configuration
+      def self.root
+	File.expand_path(File.join(Dir.pwd, '../..'))
```
Comment (cantino) [inline lib/huginn_agent/patches/coverage.rb]: Can we make these paths relative to the file using something like `File.expand_path(File.join(File.dirname(__FILE__), '../..'))`, or is the point that we don't know where this file will be?

**Comment by cantino** [comment]
I'm okay with the monkeypatch since it's only loaded in tests. One question about the paths.

File: lib/huginn_agent/patches/coverage.rb (hunk 1 above)
Comment (dsander) [inline lib/huginn_agent/patches/coverage.rb]: Actually the paths need to be relative to the current working directory. The file is required by rspec which is running in `somepath/agent_gem/spec/huginn` and we need the root to be `somepath/agent_gem`. Using `__FILE__` would result in a path somewhere in the users gem directory :)

Comment (cantino) [inline lib/huginn_agent/patches/coverage.rb]: Okay, sounds good to me then!

**Comment by dsander** [comment]
Thanks again @stesie  :)

**Pull request merged**